File: Intermediate/35.Write a program to find the intersection of two sorted arrays/Solution.py
def main():
    n1 = int(input("Enter the number of elements in the first sorted array: "))
    arr1 = list(map(int, input("Enter the elements of the first sorted array (space-separated): ").split()))
    
    n2 = int(input("Enter the number of elements in the second sorted array: "))
    arr2 = list(map(int, input("Enter the elements of the second sorted array (space-separated): ").split()))
    
    i, j = 0, 0
    intersection = []
    
    while i < n1 and j < n2:
        if arr1[i] < arr2[j]:
            i += 1
        elif arr1[i] > arr2[j]:
            j += 1
        else:
            intersection.append(arr1[i])
            i += 1
            j += 1

    if intersection:
        print("Intersection of the two arrays:", " ".join(map(str, intersection)))
    else:
        print("No common elements found.")

if __name__ == "__main__":
    main()

File: Intermediate/35.Write a program to find the intersection of two sorted arrays/Solution.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n1, n2;
    cout << "Enter the number of elements in the first sorted array: ";
    cin >> n1;
    
    vector<int> arr1(n1);
    cout << "Enter " << n1 << " sorted elements for the first array (space-separated): ";
    for (int i = 0; i < n1; i++) {
        cin >> arr1[i];
    }
    
    cout << "Enter the number of elements in the second sorted array: ";
    cin >> n2;
    
    vector<int> arr2(n2);
    cout << "Enter " << n2 << " sorted elements for the second array (space-separated): ";
    for (int i = 0; i < n2; i++) {
        cin >> arr2[i];
    }
    
    cout << "Intersection of the two arrays: ";
    int i = 0, j = 0;
    while (i < n1 && j < n2) {
        if (arr1[i] < arr2[j]) {
            i++;
        } else if (arr1[i] > arr2[j]) {
            j++;
        } else {
            cout << arr1[i] << " ";
            i++;
            j++;
        }
    }
    cout << endl;
    return 0;
}
